### PR TITLE
Add contributing docs page

### PR DESCRIPTION
**Pull request by uSpike**
The content is largely taken from HTTPX: https://github.com/encode/httpx/blob/0.18.2/docs/contributing.md

### CODE REVIEW

**Comment by graingert** [comment]
Nice the screenshots are from httpx still though ;)

**Comment by uSpike** [comment]
I think it makes sense to have some centralized documentation (or some way for projects to share docs source?) in the future so that this kind of thing doesn't need repeating.

**Comment by uSpike** [comment]
> Nice the screenshots are from httpx still though ;)

Yeah, there aren't many failing starlette PRs whos CI results haven't been tossed after expiring, so I'd have to open a dummy one to make starlette-specific images.

Alternatively I could just add the HTTPX images to this repo and not cross-link them.  IMO the images of CI don't have to be specific to starlette since both projects use the same CI setup.

**Comment by uSpike** [comment]
I added local versions of the github actions images from HTTPX.

**Review by graingert — APPROVED** [review]

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+Before releasing a new version, create a pull request that includes:
+
+- **An update to the changelog**:
+    - We follow the format from [keepachangelog](https://keepachangelog.com/en/1.0.0/).
+    - [Compare](https://github.com/encode/starlette/compare/) `master` with the tag of the latest release, and list all entries that are of interest to our users:
+        - Things that **must** go in the changelog: added, changed, deprecated or removed features, and bug fixes.
+        - Things that **should not** go in the changelog: changes to documentation, tests or tooling.
+        - Try sorting entries in descending order of impact / importance.
+        - Keep it concise and to-the-point. 🎯
+- **A version bump**: see `__version__.py`.
```
Comment (JayH5) [inline docs/contributing.md:149]: Don't think we have a `__version__.py`

File: docs/contributing.md [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+For an example, see [#1006](https://github.com/encode/starlette/pull/1202).
```
Comment (JayH5) [inline docs/contributing.md:151]: Numbers don't match

Comment (graingert) [reply docs/contributing.md:151]:
```suggestion
For an example, see [#1202](https://github.com/encode/starlette/pull/1202).
```

File: docs/contributing.md (hunk 1 above)
Comment (graingert) [reply docs/contributing.md:149]:
```suggestion
- **A version bump**: see `starlette/__init__.py`.
```

**Comment by Kludex** [comment]
@uSpike Would you mind addressing the change requests? :pray:

**Comment by Kludex** [comment]
If someone wants to continue this PR, feel free to do so. :kissing:

**Pull request closed without merging**